test-components/rpc-grpc/components-rust/rpc-grpc-api-caller/golem.yaml

```diff
@@ -0,0 +1,19 @@
+# Schema for IDEA:
+# $schema: https://schema.golem.cloud/app/golem/1.1.1/golem.schema.json
+# Schema for vscode-yaml
+# yaml-language-server: $schema=https://schema.golem.cloud/app/golem/1.1.1/golem.schema.json
+
+# See https://learn.golem.cloud/docs/app-manifest#field-reference for field reference
+
+components:
+  rpc-grpc:api-caller:
+    template: rust
+
+# Example for adding dependencies for Worker to Worker communication:
+# See https://learn.golem.cloud/docs/app-manifest#fields_dependencies for more information
+#
+dependencies:
+ rpc-grpc:api-caller:
+ - target: rpc-grpc:grpcbin
+   type: wasm-rpc
+   remote: grpc
```

Comment:
This looks wrong. For a grpc dependency, it should look like this:

```
type: grpc
proto:
  file: ../../example/service.proto
```

Reply:
Other protobuf-specific options would be nested under `proto`, for example, a target package name (although this could be inferred from the protobuf file).

Reply:
I was thinking on like remotes
dynamic rpc with different remotes like grpc, openapi, golem-worker (it uses wasmrpc)

Reply:
wit is generated manually or via customCommand.
when building cleint via build_client for dynamic wasmrpc in cli
i check for remote and generate metadata and store in application context and it used used when sending dynamic linking object. 
if it is grpc i look at grpc folder inside the component like wise other openapi remotes.

Reply:
You build support for this on wasm-rpc but wasm-rpc is only for wasm-rpc, not for gRPC. It's a different dependency type and needs to be reflected like that in the manifest file.

Now obviously you have to do many of the same steps as for wasm-rpc, but that's an implementation detail solved with code reuse / abstraction and should not affect the user-visible surface area of the feature. From a user's perspective, wasm-rpc is one dependency type, grpc is another dependency type, each requiring different things.

A grpc dependency requires a file that contains the proto.